packages/techdocs-common/src/stages/publish/azureBlobStorage.ts

```diff
@@ -139,26 +143,31 @@ export class AzureBlobStoragePublish implements PublisherBase {
           `${entityRootDir}/${relativeFilePath}`,
         ); // Azure Blob Storage Container file relative path
 
-        return limiter(async () => {
-          await uploadPromises.push(
-            this.storageClient
-              .getContainerClient(this.containerName)
-              .getBlockBlobClient(destination)
-              .uploadFile(filePath),
-          );
+        return limiter(() => {
+          return this.storageClient
+            .getContainerClient(this.containerName)
+            .getBlockBlobClient(destination)
+            .uploadFile(filePath);
         });
       });
 
-      await Promise.all(promises).then(() => {
+      let responses: BlobUploadCommonResponse[] = [];
+
+      responses = await Promise.all(promises);
+
+      const failed = responses.filter(r => r?._response?.status >= 400);
+      if (failed.length === 0) {
         this.logger.info(
           `Successfully uploaded all the generated files for Entity ${entity.metadata.name}. Total number of files: ${allFilesToUpload.length}`,
         );
-      });
-      return;
+      } else {
+        const errorMessage = `Unable to upload ${failed.length} file(s) to Azure Blob Storage.`;
+        this.logger.error(errorMessage);
+      }
     } catch (e) {
       const errorMessage = `Unable to upload file(s) to Azure Blob Storage. Error ${e.message}`;
       this.logger.error(errorMessage);
-      throw new Error(errorMessage);
+      return;
```

Comment:
We should keep the `throw new Error()` to make sure the promise returned by `publish` is rejected.